x/dogfood/keeper/abci.go

```diff
@@ -98,6 +107,7 @@ func (k Keeper) EndBlock(ctx sdk.Context) []abci.ValidatorUpdate {
 		}
 		addressString := address.String()
 		if prevPower, found := prevMap[addressString]; found && prevPower == power {
```

Comment:
for the `addressString` of which its power has changed, also need to be removed from the `preMap`